app/services/idv/phone_step.rb

```diff
@@ -1,19 +1,24 @@
 module Idv
   class PhoneStep
-    def initialize(idv_session:, trace_id:, attempts_tracker:)
+    def initialize(idv_session:, trace_id:, analytics:, attempts_tracker:)
       self.idv_session = idv_session
       @trace_id = trace_id
+      @analytics = analytics
       @attempts_tracker = attempts_tracker
     end
 
     def submit(step_params)
+      return throttled_result if throttle.throttled?
+      throttle.increment!
+
       self.step_params = step_params
       idv_session.previous_phone_step_params = step_params.slice(:phone, :otp_delivery_preference)
       proof_address
     end
 
     def failure_reason
       return :fail if throttle.throttled?
+      return if idv_result.nil?
```

Comment:
Maybe we should not return `nil` here? I suggest something like `:no_result`. We're returning a keyword in every case except that one.